rec_to_binaries/create_system_time.py

```diff
@@ -0,0 +1,109 @@
+from rec_to_binaries.read_binaries import readTrodesExtractedDataFile, write_trodes_extracted_datafile
+
+import numpy as np
+
+def add_system_time_to_file(continuous_time_file, new_file_name):
+	"""Adds system time to the continuoustime.dat file created by Trodes exporttime command.
+
+    Parameters
+    ----------
+    continuous_time_file : string, filepath to the binary file <date_animal_run>.continuoustime.dat
+    new_file_name: string, filepath to the binary output file <date_animal_run>.continuoustime.dat
+    	which contains the added continuoustime.dat to the 'data' field. Must include the
+    	'contintinuoustime.dat' in the string
+
+    Returns
+    -------
+    Writes out the complete .continuoustime.dat file specified by new_file_name
+
+    """
+
+	cont_time = readTrodesExtractedDataFile(continuous_time_file)
+	cont_time['data'] = create_and_add_system_timepoints(cont_time)
+
+	if 'fields' in cont_time:
+		cont_time['fields'] = '<trodestime uint32><systime int64>'
+	else:
+		cont_time['Fields'] = '<trodestime uint32><systime int64>'
+
+	write_trodes_extracted_datafile(new_file_name, cont_time)
+
+def create_and_add_system_timepoints(cont_time):
+	"""Creates the system time by extrapolating from the 'system_time_at_creation' timestamp
+	as a function of the 'clockrate' and the 'trodestime' containted in 'data'
+
+    Parameters
+    ----------
+    cont_time: dict, a dictionary of various fields and data created by 
+    	readTrodesExtractedDataFile(<filepath_to_continuoustime.dat_file>)
+    -------
+
+
+	Assumptions: 'clockrate' is in Hz; 'system_time_at_creation' is taken to the millisecond
+
+	""" 
+
+	NANOSECONDS_TO_SECONDS = 1e9
+
+	if 'clockrate' in cont_time:
+		nanosec_per_sample = int((1/int(cont_time['clockrate']))* (NANOSECONDS_TO_SECONDS))
+	else:
+		nanosec_per_sample = int((1/int(cont_time['Clockrate']))* (NANOSECONDS_TO_SECONDS))
+
+	sys_time_intervals = np.zeros(np.shape(cont_time['data']), dtype='int64')
+
+	if 'timestamp_at_creation' in cont_time:
+		interval = (cont_time['data'][0][0] - int(cont_time['timestamp_at_creation'])) * nanosec_per_sample
+	else:
+		interval = (cont_time['data'][0][0] - int(cont_time['Timestamp_at_creation'])) * nanosec_per_sample
+
+	
+	MILLISECONDS_TO_NANOSECONDS = 1e6
+
+	if 'system_time_at_creation' in cont_time:
+		sys_time_intervals[0] = (interval + (int(cont_time['system_time_at_creation']) * (MILLISECONDS_TO_NANOSECONDS))).astype(int)
+	else:
+		sys_time_intervals[0] = (interval + (int(cont_time['System_time_at_creation']) * (MILLISECONDS_TO_NANOSECONDS))).astype(int)
+
+	trodes_times = np.zeros(np.shape(cont_time['data']), dtype='int64') + cont_time['data'].astype(int)
+
+    trodes_intervals = trodes_times[1:] - trodes_times[:-1]
+    
+    sys_time_intervals[1:] = sys_time_intervals[1:] + (trodes_intervals * nanosec_per_sample)
+    
+    sys_time_intervals[2::3] = sys_time_intervals[2::3] + 1
```

Comment:
Can you remind me of the logic of skipping every third sample? Also will this work if the clockrate is 2000?

Reply:
Because a clockrate of 30000 equals 33333.333 nanoseconds, but I am only adding integers, so I need to compensate for rounding down by adding an additional nanosecond every 3 samples. But the 20000 clockrate is a good point, I will add a conditional.